src/store/modules/aeternity.js

```diff
@@ -118,31 +141,179 @@ export default {
         });
       });
     },
-    async tokenBalance({ state: { sdk } }, token, address) {
-      const tokenContract = await sdk
-        .getContractInstance(FUNGIBLE_TOKEN_CONTRACT, { contractAddress: token });
+    async getHeight({ state: { sdk } }) {
+      return sdk.height();
+    },
+    async initFungibleTokenContractIfNeeded(
+      { commit, state: { sdk, fungibleTokenContracts } },
+      contractAddress,
+    ) {
+      if (!fungibleTokenContracts[contractAddress]) {
+        const contract = await sdk
+          .getContractInstance(FUNGIBLE_TOKEN_CONTRACT_INTERFACE, { contractAddress });
+        commit('setFungibleTokenContract', contractAddress, contract);
+        return contract;
+      }
+
+      return fungibleTokenContracts[contractAddress];
+    },
+    async initWordRegistryContractIfNeeded({ commit, state: { sdk, wordRegistryContract } }) {
+      if (!wordRegistryContract) {
+        const contract = await sdk
+          .getContractInstance(WORD_REGISTRY_CONTRACT,
+            { contractAddress: process.env.VUE_APP_WORD_REGISTRY_ADDRESS });
+        commit('setWordRegistryContract', contract);
+        return contract;
+      }
+
+      return wordRegistryContract;
+    },
+    async initTokenVotingContractIfNeeded(
+      { commit, state: { sdk, tokenVotingContracts } },
+      contractAddress,
+    ) {
+      if (!tokenVotingContracts[contractAddress]) {
+        const contract = await sdk
+          .getContractInstance(TOKEN_VOTING_CONTRACT, { contractAddress });
+        commit('setTokenVotingContract', contractAddress, contract);
+        return contract;
+      }
+
+      return tokenVotingContracts[contractAddress];
+    },
+    async initTokenSaleContractIfNeeded(
+      { commit, state: { sdk, tokenSaleContracts } },
+      contractAddress,
+    ) {
+      if (!tokenSaleContracts[contractAddress]) {
+        const contract = await sdk.getContractInstance(TOKEN_SALE_CONTRACT, { contractAddress });
+        commit('setTokenSaleContract', contractAddress, contract);
+        return contract;
+      }
+
+      return tokenSaleContracts[contractAddress];
+    },
+    async deployBondingCurve({ state: { sdk } }, decimals) {
+      const BONDING_CURVE_DECIMALS = BONDING_CURVE.replace(
+        'function alpha() : Frac.frac = Frac.make_frac(1, 1)',
+        `function alpha() : Frac.frac = Frac.make_frac(1, ${shiftDecimalPlaces(1, decimals)})`,
+      );
```

Comment:
This works apparently, but is not intuitive what exactly it is doing, please document it.